File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        output = nums.copy()
        prefix = 1
        for x in range(len(output)):
            temp = output[x]
            output[x] = prefix
            prefix *= temp # every val in output represents the product of num before it
        postfix = 1
        for x in range(len(output)-1, -1, -1):
            output[x] *= postfix
            postfix *= nums[x]
        return output

File: Data Structures & Algorithms/concatenation-of-array/submission-1.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int[] ans = new int[nums.length * 2];
    int x = 0;
    int y = 0;
    while (x != ans.length){
        ans[x] = nums[y];
        x++;
        y++;
        if (y == nums.length){
            y -= nums.length;
        }
    }
    return ans;
}
}